File: SEM 8/Distrubuted Algorithms/U17CO110_DA_9_10/ChatApplication/java/ModeratorClient.java
import ChatApplication.*;

import org.omg.CosNaming.*;
import org.omg.CosNaming.NamingContextPackage.*;
import org.omg.CORBA.*;

import java.util.Scanner;

public class ModeratorClient {

    static Moderator moderatorImpl;
    
    public static void main(String[] args) {
        
        Scanner sc = new Scanner(System.in);
    
        try{
            // create and initialize the ORB
            ORB orb = ORB.init(args, null);
            
            // get the root naming context
            org.omg.CORBA.Object objRef = orb.resolve_initial_references("NameService");

            // Use NamingContextExt instead of NamingContext. This is 
            // part of the Interoperable naming Service.  
            NamingContextExt ncRef = NamingContextExtHelper.narrow(objRef);
            

            // resolve the Object Reference in Naming
            String name = "Moderator";
            moderatorImpl = ModeratorHelper.narrow(ncRef.resolve_str(name));

            System.out.println("[+] Connected to the moderator server: ");
            System.out.println(moderatorImpl);
            
            String username;
            System.out.print("[+] Enter your name: ");
            username = sc.nextLine();
            String reply = moderatorImpl.login(username);
            System.out.println(reply);
            
            boolean isRunning = true;

            while(isRunning){
                int choice;
                String mssg;
                String response;
                System.out.println("Choose one of below\n1.Send Mssg\n2.Fetch all messages in room\n3.Logout\n");
                choice = Integer.parseInt(sc.nextLine());

                switch(choice){
                    
                    case 1: System.out.print("[+] Enter your message: ");
                            mssg = sc.nextLine();
                            response = moderatorImpl.sendMessage(username, mssg);
                            System.out.print("\n" + response + "\n");
                            break;

                    case 2: response = moderatorImpl.getMessages();
                            System.out.print("\n" + response + "\n");
                            break;
                    
                    case 3: response = moderatorImpl.logout(username);
                            System.out.print("\n" + response + "\n");
                            isRunning = false;
                            break;

                    default: System.out.println("\n[+] No such option available\n");
                            break;
                }
            }
        } catch (Exception e) {
            System.err.println("Error: " + e);
            e.printStackTrace(System.out);
        }
    }

}

File: SEM 8/Distrubuted Algorithms/U17CO110_DA_9_10/ChatApplication/python/ModeratorClient.py
import sys
from omniORB import CORBA, PortableServer
import ChatApplication

orb = CORBA.ORB_init(sys.argv, CORBA.ORB_ID)
ior = sys.argv[1]
obj = orb.string_to_object(ior)
moderator = obj._narrow(ChatApplication.Moderator)

if moderator is None:
    print("Can't narrow reference.")
    sys.exit(1)

username = input("[+] Enter your name: ")
result = moderator.login(username)
print(result)

choice = 0
isRunning = True

while isRunning:

    choice = int(input("\nChoose any 1 of below:\n1.Send Mssg\n2.Fetch all messages in room\n3.Logout\n"));
    
    if choice == 1:
        mssg = input("[+] Enter your message: ")
        response = moderator.sendMessage(username, mssg)
        print("\n{}\n".format(response))
    elif choice == 2:
        response = moderator.getMessages()
        print("\n{}\n".format(response))
    elif choice == 3:
        response = moderator.logout(username)
        print("\n{}\n".format(response))
        isRunning = False
    else:
        print("[+] No such option.")
